### ISSUE TITLE
Predicting first ntrees with best_iteration attribute

### ISSUE THREAD

**Issue by ivallesp** [dmlc/xgboost#519]
Hello,

First of all, thanks for your job, it works amazingly good. I have an issue when I train my data with your algorithm in python... sometimes I set the num_rounds parameter too high, and it leads to overfitting, so that there is a point where the validation error begins decreasing while the training error is still growing. The early_stopping does not stop the training because the errors are still changing. Now, my question is: is there any way of choosing the best fit based on the validation error? I tried to use this line:

**bst.predict(dtest, ntree_limit=bst.best_iteration)**

but the problem is that **bst.best_iteration** gives me the best iteration based on the training evaluation. In addition, I would like to notice that of course I am using a validation set, declared using this line:

**evallist  = [(dval,'eval'), (dtrain,'train')]**

Is it possible to do what I explained?

Thank you,
Best,
Iván

**Comment by tqchen** [comment]
I think you only need to swap the order of train and eval. The early stopping is based on last set in the evallist

**Comment by ivallesp** [comment]
Wow! Now it works as I expected!! Thank you very much!
